rmw_fastrtps_shared_cpp/src/rmw_response.cpp

```diff
@@ -104,6 +104,22 @@ __rmw_send_response(
   wparams.related_sample_identity().sequence_number().low =
     (int32_t)(request_header->sequence_number & 0xFFFFFFFF);
 
+  // According to the list of possible entity kinds in section 9.3.1.2 of RTPS
+  // readers will have this bit on, while writers will not. We use this to know
+  // if the related guid is the request writer or the response reader.
+  constexpr uint8_t entity_id_is_reader_bit = 0x04;
```

Comment:
@MiguelCompany mind to a TODO for a proper fix? I opened an https://github.com/ros2/rmw_fastrtps/issues/392 to track that work.